File: frontend/admin/src/types/task.ts
export interface Task {
  id: number;
  name: string;
  providerName: string;
  taskType: string;
  schedule: string;
  parameters: Record<string, unknown>;
  isActive: boolean;
  lastRun?: string;
  nextRun?: string;
  status?: string;
}

export interface TaskFormData {
  name: string;
  providerName: string;
  taskType: string;
  schedule: string;
  isActive: boolean;
  parameters: Record<string, unknown>;
}

export type CreateTaskData = TaskFormData;
export type UpdateTaskData = Partial<TaskFormData>;

File: backend/app/modules/market/schemas/task.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict

class TaskBase(BaseModel):
    name: str
    provider_name: str
    task_type: str
    schedule: str
    is_active: bool = True
    parameters: dict[str, Any] = {}


class TaskCreateRequest(TaskBase):
    pass


class TaskUpdateRequest(TaskBase):
    pass


class TaskCreate(TaskBase):
    pass


class TaskUpdate(TaskBase):
    pass


class TaskResponse(BaseModel):
    id: int
    name: str
    description: str | None
    status: str | None
    task_type: str
    provider_name: str
    schedule: str
    is_active: bool
    parameters: dict[str, Any]
    last_run: datetime | None = None
    next_run: datetime | None = None
    model_config = ConfigDict(from_attributes=True)
